### DISCUSSION TITLE
Does this not go against the S in the Solid Principles - Single Responsibility

### DISCUSSION THREAD

**Discussion by digitlimit** [lorisleiva/laravel-actions#220] [Q&A] [answered]
Single-responsibility Principle (SRP) states:
A class should have one and only one reason to change, meaning that a class should have only one job.

When you can use a single action class as a controller, listener, command, etc, does this not go against this principle?

My second question is, can an action contain all the business logic or should I create my service class and use them in an action?
Should we have lots of code in the action?

**Comment by Wulfheart** [comment, marked as answer]
It still does one thing: Execute one part of your business logic. Everything else is just piping and plumbing around it.
